website/static/js/addProjectPlugin.js

```diff
@@ -0,0 +1,178 @@
+        };
+        self.viewState = m.prop('form'); // 'processing', 'success', 'error';
+        self.options = $.extend(self.defaults, options);
+        self.init = function _init () {
+
+        };
+        self.newProjectName = m.prop('');
+        self.newProjectDesc = m.prop('');
+        self.newProjectCategory = 'project';
+        self.goToProjectLink = m.prop('');
+        self.errorMessageType = m.prop('unknown');
+        self.errorMessage = {
+            'unknown' : 'There was an unknown error. Please try again later.'
+        };
+        self.chooseCategory = function(event){
+            self.newProjectCategory = $(this).val();
+        };
+        self.add = function _add () {
+            self.viewState('processing');
+            var url = $osf.apiV2Url('nodes/', { query : {}});
+            var data = {
+                    'data' : {
+                        'type': 'nodes',
+                        'attributes': {
+                            'title': self.newProjectName(),
+                            'category': self.newProjectCategory,
+                            'description' : self.newProjectDesc()
+                        }
+                    }
+                };
+            var success = function _success (result) {
+                self.viewState('success');
+                console.log('success', result);
+                self.goToProjectLink(result.data.links.html);
+            };
+            var error = function _error (result) {
+                self.viewState('error');
+                console.log('error', result);
+            };
+            m.request({method : 'POST', url : url, data : data, config : xhrconfig})
+                .then(success, error);
+            self.newProjectName('');
+            m.redraw(true);
+
+        };
+        self.reset = function _reset(){
+            self.newProjectName('');
+            self.viewState('form');
+            $('.modal').modal('hide');
+        };
+    },
+    view : function (ctrl, options) {
+        var templates = {
+            form : m('.modal-content', [
+                m('.modal-header', [
+                    m('button.close[data-dismiss="modal"][aria-label="Close"]',{ onclick : ctrl.reset}, [
+                        m('span[aria-hidden="true"]','×'),
+                    ]),
+                    m('h3.modal-title#addProject', 'Add New Project')
+                ]),
+                m('.modal-body', [
+                    m('', [
+                        m('.form-group.m-v-sm', [
+                            m('label[for="projectName].f-w-lg.text-bigger', 'Project Name'),
+                            m('input[type="text"].form-control#projectName', {
+                                onkeyup: function(ev){
+                                    if (ev.which === 13) {
+                                         ctrl.add();
+                                    }
+                                    ctrl.newProjectName($(this).val());
```

Comment:
What is `this` in this context? It seems most straightforwared to do `ev.value`.